src/search_dragon/external_apis/umls_api.py

```diff
@@ -0,0 +1,221 @@
+"""
+UMLS (Unified Medical Language System) API Integration
+
+This script defines the `UMLSSearchAPI` class that interacts with the umls API to perform ontology searches. The class provides methods to:
+- Construct search URLs with query parameters.
+- Fetch paginated search results from the UMLS API.
+- Harmonize and structure raw results into a standardized format.
+
+"""
+
+from search_dragon.external_apis import OntologyAPI
+from search_dragon import logger
+
+UMLS_API_BASE_URL = "https://uts-ws.nlm.nih.gov/rest/search/current"
```

Comment:
For these UMLS specific variables, you could define them as part of the class itself (so define them immediately after the class definition you have on line #19). Then, when you refer to them, you would refer to them like: UMLSSearchAPI.base_url or whatever. This helps aggregate those details more closely. Or, if they are actually present in every single class, but different, they can be assigned using the base class __init__() but each class passes the appropriate values. Then, they would be referred to using self.base_url, etc.